### DISCUSSION TITLE
Support for course audio?

### DISCUSSION THREAD

**Discussion by jakevastola** [kantord/LibreLingo#2329] [General]
Hello!
I've looked through [Sourcing audio from Lingua Libre #1464 ](https://github.com/LibreLingo/LibreLingo/issues/1464) and [Import audio data from other free/open-source projects #714](https://github.com/LibreLingo/LibreLingo/issues/714).
After downloading all external courses, I notice there are audio files in:
> LibreLingo>apps>web>static>voice

Questions:
1. Where are these audio files found in the sample Spanish for English speakers course? Are they even used?
2. Is it possible to provide audio for whole sentences and individual words in my own course? If so, how can I do this? I can provide my own audio.

Thank you in advance!

**Comment by kantord** [comment]
> Where are these audio files found in the sample Spanish for English speakers course? Are they even used?

I'm not sure I understand the question, but the audio files are generated using TTS and they are used in the course. They should be available in practically every skill of the Spanish course

**Comment by kantord** [comment]
> Is it possible to provide audio for whole sentences and individual words in my own course? If so, how can I do this? I can provide my own audio.

Right now I would recommend uploading them to Tatoeba and/or Lingua Libre with some nice license. Then people can start to use them immediately while we implement the way to automatically integrate them in LibreLingo

**Reply by jakevastola** [reply]
Thank you for the response. I suppose I can look into uploading recordings of words and sentences on Tatoeba for now. Hopefully, LibreLingo will support audio for individual words, whole sentences, and clicking on pictures. I would like if LibreLingo had a feature implemented where one could click on the word to hear it out loud or click on a button to hear the entire sentence (and a button for a "slow mode" version of a recording). I suppose these are all for possibilities for the future.

**Reply by kantord** [reply]
yes, those features are all wanted and should be supported
